Orange/widgets/data/owtranspose.py

```diff
@@ -57,20 +54,23 @@ def _feature_combo_changed(self):
         self.apply()
 
     def set_data(self, data):
-        self.closeContext()
+        # Skip the context if the combo is empty: a context with
+        # feature_model == None would then match all domains
+        if self.feature_model:
+            self.closeContext()
         self.data = data
         self.update_controls()
-        self.openContext(data)
+        if self.data is not None and self.feature_model:
+            self.openContext(data)
         self.apply()
 
     def update_controls(self):
         self.feature_model.set_domain(None)
         if self.data:
             self.feature_model.set_domain(self.data.domain)
-            if len(self.feature_model):
-                _names = [m.name for m in self.data.domain.metas if m.is_string]
```

Comment:
The beauty of the domain model is that you don't need to do this. Besides, it should use `m`, not `m.name`.